contracts/adapters/swap/astroport/src/contract.rs

```diff
@@ -211,14 +212,18 @@ fn execute_astroport_pool_swap(
     env: Env,
     info: MessageInfo,
     operation: SwapOperation,
+    offer_asset: Option<Asset>,
 ) -> ContractResult<Response> {
     // Ensure the caller is the contract itself
     if info.sender != env.contract.address {
         return Err(ContractError::Unauthorized);
     }
 
     // Get the current asset available on contract to swap in
-    let offer_asset = get_current_asset_available(&deps, &env, &operation.denom_in)?;
+    let offer_asset = match offer_asset {
+        Some(offer_asset) => offer_asset,
+        None => get_current_asset_available(&deps, &env, &operation.denom_in)?,
```

Comment:
I believe there's an edge case here, ex:
1. 100 OSMO in, 2 route trade, each route has an in of 50 OSMO
2. First route is: (Osmo->Atom, Atom->Stars, Stars->Osmo, Osmo->USDC)
3. That last operation (Osmo->USDC) would use all the available Osmo asset, leaving none for the 2nd route?

The only real reason why a certain route would be given (instead of just the last osmo->usdc) is if the prior operations net more Osmo and that's why it's the better swap route. So either we want to disallow these types of routes in validation or we will need a diff way to do accounting (accounting differences in balance after each swap)

Reply:
damn I didn't know we did routes like this, ok I'll handle this differently